src/cursor/cursor.ts

```diff
@@ -158,34 +176,21 @@ export class Cursor extends CoreCursor {
     if (this.options.noCursorTimeout === true) {
       this.addCursorFlag('noCursorTimeout', true);
     }
-
-    // Get the batchSize
-    let batchSize = 1000;
-    if (this.cmd.cursor && this.cmd.cursor.batchSize) {
-      batchSize = this.cmd.cursor.batchSize;
-    } else if (options.cursor && options.cursor.batchSize) {
-      batchSize = options.cursor.batchSize;
-    } else if (typeof options.batchSize === 'number') {
-      batchSize = options.batchSize;
-    }
-
-    // Set the batchSize
-    this.setCursorBatchSize(batchSize);
   }
 
-  get readPreference() {
+  get readPreference(): ReadPreferenceLike | undefined {
```

Comment:
this should always return a `ReadPreference`, not a `ReadPreferenceLike` and definitely not `undefined`